shared/utils.js

```diff
@@ -0,0 +1,176 @@
+'use strict';
+
+const constants = {
+  type: 'type',
+  direction: 'direction',
+  trigger: 'Trigger',
+  inDirection: 'in',
+  outDirection: 'out',
+  settings: 'settings',
+  name: 'name',
+  value: 'value',
+  resource: 'resource',
+  required: 'required',
+  storage: 'storage',
+  connection: 'connection',
+  enum: 'enum',
+  defaultValue: 'defaultValue',
+  webHookType: 'webHookType',
+  httpTrigger: 'httpTrigger',
+  queue: 'queue',
+  queueName: 'queueName',
+  displayName: 'displayName',
+  xAzureSettings: 'x-azure-settings',
+  entryPoint: 'entryPoint'
+};
+
+module.exports = {
+  getFunctionMetaData: function (functionName, parsedBindings, serverless) {
+    var bindings = [];
+    var bindingSettingsNames = [];
+    var bindingSettings = [];
+    var functionsJson = { 'disabled': false, 'bindings': [] };
+    const functionObject = serverless.service.getFunction(functionName);
+    var handler = functionObject.handler;
+    var events = functionObject.events;
+    const params = {};
+
+    var bindingTypes = parsedBindings['bindingTypes'];
+    var bindingDisplayNames = parsedBindings['bindingDisplayNames'];
+
+    for (var eventsIndex = 0; eventsIndex < events.length; eventsIndex++) {
+      var bindingType = Object.keys(functionObject.events[eventsIndex])[0];
+      if (eventsIndex === 0) {
+        bindingType = bindingType + constants.trigger;
+      }
+
+      var index = bindingTypes.indexOf(bindingType);
+      if (index < 0) {
+        throw new Error(`Binding  ${bindingType} not supported`);
+      }
+
+      serverless.cli.log(`Building binding for function: ${functionName} event: ${bindingType}`);
+
+      var bindingUserSettings = {};
+      var azureSettings = events[eventsIndex][constants.xAzureSettings];
+      var bindingTypeIndex = bindingTypes.indexOf(bindingType);
+      bindingTypeIndex = this.setBindingDirection(azureSettings, bindingType, bindingDisplayNames, bindingTypeIndex, bindingUserSettings);
+
+      if (bindingType.includes(constants.queue) && functionObject.events[eventsIndex].queue) {
+        bindingUserSettings[constants.queueName] = functionObject.events[eventsIndex].queue;
+      }
+
+      if (bindingTypeIndex < 0) {
+        throw new Error('Binding not supported');
+      }
+
+      bindingSettings = parsedBindings['bindingSettings'][bindingTypeIndex];
+      bindingSettingsNames = parsedBindings['bindingSettingsNames'][bindingTypeIndex];
+
+      if (azureSettings) {
+        for (var j = 0; j < Object.keys(azureSettings).length; j++) {
```

Comment:
Loops should use `let` instead of var unless you mean to use the variable outside the block scope